src/blmc_joint_module.cpp

```diff
@@ -22,7 +22,7 @@ BlmcJointModule::BlmcJointModule(std::shared_ptr<blmc_drivers::MotorInterface> m
                     const double& zero_angle,
                     const bool& reverse_polarity,
                     const double& max_current):
-    homing_state_({0})
+    homing_state_(HomingState())
```

Comment:
Maybe I am wrong, but I think this will not initialise the members to zero.  If this is indeed not the case, please change it back.

I know that this causes brace-initialiser warnings but unless there is another easy way to initialise a complete struct to zero, I would actually suggest to disable that warning.

Reply:
Simple things to do, create a constructor in your struct. Which I think it is anyway a good practice to ensure the correct default behavior.
I will add such constructor if you want.

Reply:
Alright, however, instead of a constructor I would prefer to directly define the members with default values like this:
```
struct Foo {
    int a = 0;
    int b = 0;
};
```
This is IMO more readable and makes it less likely to miss the initialisation when adding new members (however, that risk is still there, which is why I like `= {0}`. I think it is just a flaw of the compiler to issue warnings in this case...).

When going this way, that line could actually completely be removed as `homing_state_(HomingState())` is the default anyway.

Reply:
Fair point! I will try the default init per variable then to see if the compiler issue warnings.